### PR TITLE
Doc updates for ingest classes

### PR DESCRIPTION
**Pull request by apmoriarty**
Update docs for the following classes
- AbstractContentIngestHelper
- BaseIngestHelper
- CompositeIngest
- DataTypeHelper
- DataTypeHelperImpl
- IngestHelperInterface
- VirtualIngest

### CODE REVIEW

File: warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/BaseIngestHelper.java [hunk 1]
```diff
@@ -244,7 +244,7 @@ public void setup(Configuration config) {
         // Load the proper list of fields to (not) index
         if (fieldConfigHelper != null) {
             log.info("Using field config helper for " + this.getType().typeName());
-        } else if (null == configProperty || configProperty.isEmpty()) {
```
Comment (apmoriarty) [inline warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/BaseIngestHelper.java:247]: ` || configProperty.isEmpty()` is redundant

File: warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/BaseIngestHelper.java [hunk 2]
```diff
@@ -288,7 +288,7 @@ public void setup(Configuration config) {
         }
         
         // Load the proper list of fields to (not) reverse index
-        if (null == configProperty || configProperty.isEmpty()) {
```
Comment (apmoriarty) [inline warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/BaseIngestHelper.java:291]: `|| configProperty.isEmpty()` is redundant

File: warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/CompositeIngest.java [hunk 3]
```diff
@@ -145,7 +145,7 @@ public void setup(Type type, Configuration config) {
                                     .collect(Collectors.toList());
                     
                     // if any members are indexOnly fields, skip this one
-                    if (Sets.intersection(Sets.newHashSet(componentFields), indexOnly).size() > 0) {
```
Comment (apmoriarty) [inline warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/CompositeIngest.java:148]: replace `size() > 0` with `isEmpty()`

File: warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/VirtualIngest.java [hunk 4]
```diff
@@ -674,11 +716,9 @@ private VirtualFieldGrouping getGrouping(NormalizedContentInterface value, Group
         }
         
         private VirtualFieldGrouping getGrouping(NormalizedContentInterface value) {
-            if (value instanceof GroupedNormalizedContentInterface) {
-                if (((GroupedNormalizedContentInterface) value).isGrouped()) {
-                    return new VirtualFieldGrouping(((GroupedNormalizedContentInterface) value).getGroup(),
-                                    ((GroupedNormalizedContentInterface) value).getSubGroup());
-                }
```
Comment (apmoriarty) [inline warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/VirtualIngest.java:681]: do not cast values in constructor args

**Review by FineAndDandy — APPROVED** [review]

**Review by ivakegg — APPROVED** [review]

**Review by hlgp — APPROVED** [review]

File: warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/AbstractContentIngestHelper.java [hunk 5]
```diff
@@ -45,14 +48,15 @@ public abstract class AbstractContentIngestHelper extends BaseIngestHelper {
      * designator if any is specified.
      *
      * @param field
-     * @return
+     *            the field
+     * @return true if the field value should be indexed as a list of values
```
Comment (matthpeterson) [inline warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/AbstractContentIngestHelper.java:52]: maybe "reverse" indexed

**Review by matthpeterson — APPROVED** [review]

**Pull request merged**